src/Kestrel.Core/Internal/AddressBinder.cs

```diff
@@ -43,7 +43,7 @@ internal class AddressBinder
             listenOptions.Clear();
             addresses.Addresses.Clear();
 
-            await strategy.BindAsync(context).ConfigureAwait(false);
+            return strategy.BindAsync(context);
```

Comment:
Nope, all code in the startup path needs ConfigureAwait. Also there's no point "optimizing" this code path since it's server startup.

Reply:
isnt ConfigureAwait only relevant if the task is awaited in this context? if it is not awaited no state machine is created, and it is up to the caller to choose to ConfigureAwait or not?

Reply:
Yes, that's true. Still I see no value in removing this one only. Usually, you'd make it so that you need to get off the sync context once and all lower calls can avoid the calls to ConfigureAwait, but this really doesn't save anything.

Reply:
a little confused. this is the recommended way of handling a single async call, it is a little bit faster (yes at startup but still), it is just as readable. So why not accept the PR?

Also why is this change bad but the one in the test is ok?

Reply:
> this is the recommended way of handling a single async call

The recommendations for these types of things are extremely fuzzy. This changes the behavior (exceptions are now not faulted tasks but they can throw synchronously), it's of the method and there's no perf gain to be had. This is called once per application.

The removal of the ConfigureAwait when all of the calls underneath it have ConfigureAwait also doesn't sit well with me. Subtle changes like this that affect the behavior when there's no real performance gain is what I think there's no point making this change here.

Reply:
> Also why is this change bad but the one in the test is ok?

The test method has a single line of code. This method has lots of code and ends with a tail call. They are different.